### ISSUE TITLE
Enhance fileZipOperation to zip a list of files and folders

### ISSUE THREAD

**Issue by majdi21** [jenkinsci/file-operations-plugin#133]
### What feature do you want to see added?

Hello,

  I want to be able to zip a list of files and folder, not just a folder. Here are some details.

  Say I have a folder called **contents**, with a subfolder called **files**. Under **files**, I have the following files **a.txt**, **b.txt**, and **c.txt**. I also have a file **info.xml** under the folder **contents**. Here is a visual of that structure:

contents
  | - info.xml
  |- files
      |- a.txt
      |- b.txt
      |- c.txt

When I use fileZipOperation, I have to specify a folder which means I have to specify the folder **contents**. This results in a zip file containing the directory **contents** and all of its contents.

But I want to zip the contents of the **contents** folder without having the **contents** folder in the zip file.

In other words instead of this:

**package.zip**
  |-contents
      | - info.xml
      |- files
          |- a.txt
          |- b.txt
          |- c.txt

I want the zip file to look like this:

**package.zip**
  | - info.xml
  |- files
      |- a.txt
      |- b.txt
      |- c.txt

**Comment by mohit-uchit** [comment]
Yes , its easy . I am interested . May you please give the use case . Meanwhile if you wanna do only for a single case you can use the following code : 

```
const fs = require('fs');
const path = require('path');
const archiver = require('archiver');

function zipFolderContents(folderPath, outputZipPath) {
    const output = fs.createWriteStream(outputZipPath);
    const archive = archiver('zip', {
        zlib: { level: 9 }
    });

    output.on('close', () => {
        console.log(`Zipped ${archive.pointer()} total bytes.`);
        console.log('Zip file has been created successfully.');
    });

    archive.on('error', (err) => {
        throw err;
    });

    archive.pipe(output);

    fs.readdir(folderPath, (err, files) => {
        if (err) {
            throw err;
        }

        files.forEach(file => {
            const filePath = path.join(folderPath, file);
            archive.append(fs.createReadStream(filePath), { name: file });
        });

        archive.finalize();
    });
}

const folderToZip = path.join(__dirname, 'contents'); // path to the 'contents' folder
const outputZipFile = path.join(__dirname, 'package.zip');

zipFolderContents(folderToZip, outputZipFile);

```
Thanks

**Comment by majdi21** [comment]
@jonesbusy A day after I posted this request, I got an email from "Mohit Uchit" on this issue. It looked like a normal reply. But I can't see the reply on the issue. I'm not sure why. Maybe Mohit deleted the response? Any idea?

**Comment by jonesbusy** [comment]
I'm not sure. I'm not activeley adding feature on this plugin. Just to keep it updated and review contributions if any

**Comment by Mahesh-Sivashankaran** [comment]
@majdi21 @jonesbusy 
But by default, when you try to Zip a folder or some files, you will have the outer folder, Here its `contents`
But instead of removing the folder while zipping
We can move one step up in folder dir while unzipping

**Comment by Mahesh-Sivashankaran** [comment]
I also wish to contribute to the same.
